AEPCore/Sources/eventhub/EventHub.swift

```diff
@@ -86,6 +87,11 @@ final class EventHub {
             self?.eventQueue.add(event)
             Log.trace(label: self?.LOG_TAG ?? "EventHub",
                       "Dispatching Event #\(String(describing: self?.eventNumberMap[event.id] ?? 0)) - \(event)")
+
+            // record the event in history if it has a mask
+            if event.mask != nil {
+                self?.eventHistory?.recordEvent(event)
```

Comment:
We'd better add a debug log here if the `EventHistory` is not initialized correctly.